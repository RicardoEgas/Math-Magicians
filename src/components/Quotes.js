import { useState, useEffect } from 'react';
import './styles.css';

const Quote = () => {
  const [data, setData] = useState([{ author: '', quote: '', category: '' }]);
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const apiKey = 'gzO3VhbHDwzrcSjHefdhdw==HJjjOoRw21MM6Yc3';
        const apiUrl = 'https://api.api-ninjas.com/v1/quotes';
        const response = await fetch(apiUrl, {
          headers: {
            'Content-Type': 'application/json',
            'X-Api-Key': apiKey,
          },
        });
        const data = await response.json();
        setData(data[0]);
      } catch (error) {
        setError(error.message);
      }
      setLoading(false);
    };
    fetchData();
  }, []);

  if (error) return <div>Something went wrong!</div>;

  if (loading) return <div>Loading...</div>;

  return (
    <div className="container-quotes">
      <div className="quote-text">
        <span>
          &quot;
          {data.quote}
          &quot;
        </span>
      </div>
      <div className="quote-author">
        <span>
          {' '}
          By
          {' '}
          {data.author}
        </span>
      </div>
    </div>
  );
};

export default Quote;
