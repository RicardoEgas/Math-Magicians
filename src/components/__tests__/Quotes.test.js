import React from 'react';
import { render, waitFor } from '@testing-library/react';
import Quotes from '../Quotes';

describe('Quote component', () => {
  it('should match the snapshot', async () => {
    const { container } = render(<Quotes />);

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
