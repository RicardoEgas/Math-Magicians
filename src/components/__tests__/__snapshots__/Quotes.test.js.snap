// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component should match the snapshot 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
